extensions/ql-vscode/src/codeql-cli/distribution.ts

```diff
@@ -475,19 +494,41 @@ class ExtensionSpecificDistributionManager {
   }
 
   private createReleasesApiConsumer(): ReleasesApiConsumer {
-    const ownerName = this.config.ownerName
-      ? this.config.ownerName
-      : DEFAULT_DISTRIBUTION_OWNER_NAME;
-    const repositoryName = this.config.repositoryName
-      ? this.config.repositoryName
-      : DEFAULT_DISTRIBUTION_REPOSITORY_NAME;
     return new ReleasesApiConsumer(
-      ownerName,
-      repositoryName,
+      this.distributionOwnerName(),
+      this.distributionRepositoryName(),
       this.config.personalAccessToken,
     );
   }
 
+  private distributionOwnerName(): string {
+    if (this.config.ownerName) {
+      return this.config.ownerName;
+    } else if (this.config.channel === "nightly") {
+      return NIGHTLY_DISTRIBUTION_OWNER_NAME;
+    } else {
+      return DEFAULT_DISTRIBUTION_OWNER_NAME;
+    }
+  }
+
+  private distributionRepositoryName(): string {
+    if (this.config.repositoryName) {
+      return this.config.repositoryName;
+    } else if (this.config.channel === "nightly") {
+      return NIGHTLY_DISTRIBUTION_REPOSITORY_NAME;
+    } else {
+      return DEFAULT_DISTRIBUTION_REPOSITORY_NAME;
+    }
+  }
+
+  private usingNightlyReleases(): boolean {
```

Comment:
Could this perhaps be based on matching the distribution owner and repository name to `dsp-testing/codeql-cli-nightlies`? That might be a bit easier and just changes the behavior if you manually set it to this repository before this change. That would have given an error before, and would now start working.

Reply:
That's a good idea. I'll change this is to check if the value matches the default value.